lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -703,11 +712,21 @@ def _on_config_changed(self, event: ConfigChangedEvent):  # noqa C901
             # config-changed is called again.
             if self.unit.is_leader():
                 self.status.clear(PluginConfigCheck, app=True)
-            return
+        else:
+            if self.unit.is_leader():
+                self.status.clear(PluginConfigCheck, app=True)
+                self.status.clear(PluginConfigChangeError, app=True)
 
-        if self.unit.is_leader():
-            self.status.clear(PluginConfigCheck, app=True)
-            self.status.clear(PluginConfigChangeError, app=True)
+        if deployment_desc.typ == DeploymentType.MAIN_ORCHESTRATOR:
+            # Only the main orchestrator can change the performance profile
+            perf_profile_needs_restart = self.performance_profile.apply(
+                self.config.get(PERFORMANCE_PROFILE)
+            )
+            if self.opensearch_peer_cm.is_provider(typ="main") and perf_profile_needs_restart:
```

Comment:
@Mehdi-Bendriss do you think we should gate the `refresh_relation_data` with `self.opensearch_peer_cm.is_provider(typ="main")`. Effectively, that means the `DeploymentDescription` only gets updated at when the first orch. requirer shows up.